Add `RGBLIGHT_DEFAULT_ON` macro configuration option (#20857)
qmk/qmk_firmware: docs/feature_rgblight.md

|`RGBLIGHT_DEFAULT_SAT`     |`UINT8_MAX` (255)           |The default saturation to use upon clearing the EEPROM                                                                     |
|`RGBLIGHT_DEFAULT_VAL`     |`RGBLIGHT_LIMIT_VAL`        |The default value (brightness) to use upon clearing the EEPROM                                                             |
|`RGBLIGHT_DEFAULT_SPD`     |`0`                         |The default speed to use upon clearing the EEPROM                                                                          |

> |`RGBLIGHT_DEFAULT_ON`      |`true`                      |Enable RGB lighting upon clearing the EEPROM                           |

Reviewer: ```suggestion
|`RGBLIGHT_DEFAULT_ON`      |`true`                      |Enable RGB lighting upon clearing the EEPROM                                                                               |
```

> |`RGBLIGHT_DEFAULT_ON`      |`1`                         |If defined  `0`, RGB lighting will be disabled (off) by default when EEPROM is initialized.                                |

Reviewer: ```suggestion
|`RGBLIGHT_DEFAULT_ON`      |`true`                      |Enable RGB lighting upon clearing the EEPROM                           |
```
